src/js/common/schemaValidatedModelComponent.js

```diff
@@ -16,17 +16,24 @@ var fluid  = fluid  || {};
      *
      * Validate the model against the associated schema (options.modelSchema).
      *
-     * @param {Object} that - The component itself.
+     * @param {Object} globalValidator - The global validation component.
+     * @param {Object} modelValidationComponent - The component itself.
      *
      */
-    gpii.schema.modelComponent.validateModel = function (that) {
-        var validationResults = gpii.schema.validator.validate(that.model, that.options.modelSchema);
+    gpii.schema.modelComponent.validateModel = function (globalValidator, modelValidationComponent) {
+        if (!modelValidationComponent.schemaHash) {
+            modelValidationComponent.schemaHash = gpii.schema.hashString(gpii.schema.stringify(modelValidationComponent.options.modelSchema));
+        }
+        var validationResults = globalValidator.validate(modelValidationComponent.options.modelSchema, modelValidationComponent.model, modelValidationComponent.schemaHash);
         // Flag this change as a result of validation so that we can avoid multiple validation passes per model change.
-        that.applier.change("validationResults", validationResults, "ADD", "validation");
+        modelValidationComponent.applier.change("validationResults", validationResults, "ADD", "validation");
     };
 
     fluid.defaults("gpii.schema.modelComponent", {
         gradeNames: ["gpii.schema.component", "fluid.modelComponent"],
+        members: {
+            schemaHash: false
```

Comment:
Why not generate the schema hash in an expander in place?

Reply:
My goal was to make it easier to correlate a cached schema with its original component.  If we don't want to use more readable hashes,  I'm not sure there's much value in storing the stringified schema in the component and passing it around.